### PR TITLE
Create unique instance for each AutoComplete object

### LINKED ISSUE #6953: Automation case about comboBox met an error "Object reference not set to an instance of an object"
### .NET version

.Net 7.0.100-preview.4.22201.11

### Did it work in .NET Framework?

No

### Did it work in any of the earlier releases of .NET Core or .NET 5+?

Yes, cannot repro in .Net SDK 7.0 Preview3 build

### Issue description

Automation case about comboBox met an error "Object reference not set to an instance of an object":
image

### Steps to reproduce

We cannot get a manual repro, just met it when running our internal automation case about ComboBox, get following calll stack:

_System.NullReferenceException
HResult=0x80004003
Message=Object reference not set to an instance of an object.
Source=System.Windows.Forms.Primitives
StackTrace:
at Interop.WinFormsComWrappers.AutoCompleteWrapper.SetOptions(AUTOCOMPLETEOPTIONS dwFlag)
at System.Windows.Forms.StringSource.Bind(HandleRef edit, AUTOCOMPLETEOPTIONS options)
at System.Windows.Forms.ComboBox.SetAutoComplete(Boolean reset, Boolean recreate)
at MMFComboBox.CustomSource(TParams p) in C:\Users\v-olzhan\Desktop\Automation\MMFComboBox_P1\MMFComboBox_P1\MMFComboBox\MMFComboBox.cs:line 624_

### PR DESCRIPTION
**Pull request by kant2002**
That way we do not re-use released RCW

Fixes #6953

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6958)

### CODE REVIEW

**Comment by kant2002** [comment]
@RussKie or @dreddy-work please advice on what I should also do? if I can create test out of ugly repro, where should I place it

**Comment by RussKie** [comment]
[Your sample](https://github.com/dotnet/winforms/issues/6953#issuecomment-1086862991) looks like it can be added to https://github.com/dotnet/winforms/tree/main/src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests project. I suggest creating a new `ComboBoxTests` class, and add your sample there. A test case could be as simple as `ComboBoxTest_ChangeAutoCompleteSource_DoesntThrow` (checkout the [naming rules](https://github.com/dotnet/winforms/blob/main/docs/testing.md#naming)).
Does this make sense?

**Comment by kant2002** [comment]
Yes. I add test and verify that it fails on the main branch. After fix it's passing :)

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+                "_sss"
+            });
+            comboBox.AutoCompleteSource = AutoCompleteSource.CustomSource;
+            comboBox.AutoCompleteSource = AutoCompleteSource.CustomSource;
+            comboBox.AutoCompleteMode = AutoCompleteMode.Suggest;
+            comboBox.AutoCompleteMode = AutoCompleteMode.Suggest;
+            comboBox.AutoCompleteMode = AutoCompleteMode.Suggest;
+
+            // Allow a little time to process the keystroke.
+            await Task.Delay(DelayMS);
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs:39]: The issue manifests without delays. In general we should avoid this (https://github.com/dotnet/winforms/pull/6540#discussion_r810358464) as tests are run on different hardware (including VMs).
```suggestion

                return Task.CompletedTask;
```

**Review by RussKie — COMMENTED** [review]
👍 with a suggestion

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+    public ComboBoxTests(ITestOutputHelper testOutputHelper)
+        : base(testOutputHelper)
+    {
+    }
+
+    [WinFormsFact]
+    public async Task ComboBoxTest_ChangeAutoCompleteSource_DoesntThrowAsync()
+    {
+        await RunSingleControlTestAsync<ComboBox>(async (form, comboBox) =>
```
Comment (kant2002) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs:23]:
```suggestion
        await RunSingleControlTestAsync<ComboBox>((form, comboBox) =>
```

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+                "_sss"
+            });
+            comboBox.AutoCompleteSource = AutoCompleteSource.CustomSource;
+            comboBox.AutoCompleteSource = AutoCompleteSource.CustomSource;
+            comboBox.AutoCompleteMode = AutoCompleteMode.Suggest;
+            comboBox.AutoCompleteMode = AutoCompleteMode.Suggest;
+            comboBox.AutoCompleteMode = AutoCompleteMode.Suggest;
+
+
+                return Task.CompletedTask;
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs:39]: incorrect indent and extra blank line :)

File: src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+using Xunit;
+using Xunit.Abstractions;
+
+namespace System.Windows.Forms.UITests;
+
+public class ComboBoxTests : ControlTestBase
+{
+    // This value may need to be adjusted if tests fail in CI/different environment.
+    private const int DelayMS = 100;
```
Comment (RussKie) [inline src/System.Windows.Forms/tests/IntegrationTests/UIIntegrationTests/ComboBoxTests.cs:13]: This should go too.

**Review by RussKie — APPROVED** [review]

**Comment by RussKie** [comment]
Thank you

**Pull request merged**